src/lexer.c

```diff
@@ -257,22 +257,45 @@ void tryParseImportStatement () {
         break;
     case '"':
     case '\'':
-    case '{':
-    case '*':
+    case '*': {
       // import statement only permitted at base-level
       if (openTokenDepth != 0) {
         pos--;
         return;
       }
       while (pos < end) {
         ch = *pos;
-        if (ch == '\'' || ch == '"') {
+        if (isQuote(ch)) {
+          readImportString(startPos, ch);
+          return;
+        }
+        pos++;
+      }
+      syntaxError();
+      break;
+    }
+    
+    case '{': {
+      if (openTokenDepth != 0) {
+        pos--;
+        return;
+      }
+      bool end_curly_bracket = false;
+      while (pos < end) {
+        ch = *pos;
+        if (ch == '}') {
```

Comment:
We need to ensure we parse the quote strings fully here - otherwise this will fail on:

```
import { asdf as "b} from 'wrong'" } from 'right';
```